[19780] Add interface whitelist by name (#591)
eProsima/Fast-DDS-docs: docs/fastdds/transport/whitelist.rst

Using *Fast DDS*, it is possible to limit the network interfaces used by :ref:`transport_tcp_tcp` and
:ref:`transport_udp_udp`.
This is achieved by adding the interfaces to the |SocketTransportDescriptor::interfaceWhiteList-api|
field in the :ref:`transport_tcp_transportDescriptor` or :ref:`transport_udp_transportDescriptor`.
Thus, the communication interfaces used by the |DomainParticipants| whose |TransportDescriptorInterface-api| defines an
|SocketTransportDescriptor::interfaceWhiteList-api| is limited to the interfaces' addresses defined in that list,
therefore avoiding the use of the rest of the network interfaces available in the system.

> It is possible to add the interfaces to |SocketTransportDescriptor::interfaceWhiteList-api| specifying the IP addresses
> or the interface names.

Reviewer: ```suggestion
The interfaces in |SocketTransportDescriptor::interfaceWhiteList-api| can be specified both by IP address
or interface name.
```

For example:

* Interface whitelist filled with IP address:

> 

Reviewer: Indent the tabs so they are nested withing the bullet list

      :lines: 2-3,5-
      :append: </profiles>

* Interface whitelist filled with interface names:

> .. tabs::

Reviewer: Same as before, indent the tabs

  .. tab:: C++

    .. literalinclude:: /../code/DDSCodeTester.cpp
      :language: c++
      :start-after: //WHITELIST-NAME
      :end-before: //!--
      :dedent: 8

  .. tab:: XML

    .. literalinclude:: /../code/XMLTester.xml
      :language: xml
      :start-after: <!-->WHITELIST-NAME
      :end-before: <!--><-->
      :lines: 2-3,5-
      :append: </profiles>

.. important::

>   If none of the values in the transport descriptor's whitelist match the interfaces on your machine,
>   all the interfaces in the whitelist are filtered out and no communication will be established through that transport.

Reviewer: ```suggestion
  If none of the values in the transport descriptor's whitelist matches the interfaces present on the host,
  then all the interfaces in the whitelist are filtered out and therefore no communication will be established through that transport.
```

      :lines: 2-3,5-
      :append: </profiles>

* Interface whitelist filled with interface names:

.. tabs::

  .. tab:: C++

    .. literalinclude:: /../code/DDSCodeTester.cpp
      :language: c++
      :start-after: //WHITELIST-NAME
      :end-before: //!--
      :dedent: 8

  .. tab:: XML

    .. literalinclude:: /../code/XMLTester.xml
      :language: xml
      :start-after: <!-->WHITELIST-NAME
      :end-before: <!--><-->
      :lines: 2-3,5-
      :append: </profiles>

.. important::

>   The values in this list should match the IP addresses or interface names on your machine for the corresponding networks. If none of the values in the whitelist match the interfaces on your machine, all the interfaces in the whitelist are filtered out and an invalid interface is assigned.

Reviewer 2: ```suggestion
  If none of the values in the transport descriptor's whitelist match the interfaces on your machine, all the interfaces in the whitelist are filtered out and no communication will be established through that transport.
```

Reviewer 2: Just to shorten it a bit, and saying we add an invalid ip is perhaps too low level for a user.
